### ISSUE TITLE
[Development] Implement SpotPy solver into Pastas

### ISSUE THREAD

**Issue by raoulcollenteur** [pastas/pastas#447]
**Describe the proposed feature**

I propose to add SpotPy as an optional solver to Pastas. SpotPy "_is a Python framework that enables the use of Computational optimization techniques for calibration, uncertainty and sensitivity analysis techniques of almost every (environmental-) model._". It has many algorithms and has been applied in hydrology a lot. It is also actively maintained. 
 
I also found a victim to help with the implementation into a Pastas solver, the great @mvremec who has a lot of experience with the package :)

**Expected usage**

Here's how I imagine the solver to be used (surprise :p)

`ml.solve(solver=ps.SpotpySolve, **options)`

where the options are passed on to the solver. That means a `SpotpySolve` class needs to be constructed and added to solver.py. 

**Additional context**

- https://spotpy.readthedocs.io/en/latest/
- https://github.com/thouska/spotpy

**Comment by raoulcollenteur** [comment]
@mvremec, did you already create a branch for the solver?

**Comment by raoulcollenteur** [comment]
After initial tries (see the branch), we got SpotPy working. However, after taking a closer look at the code of Spotpy we decided to not further pursue this because the spotpy packages is not well enough developed for practical use other than science.

- spotpy still uses print statements, so you get annoying messages all the time
- there are no API docs, making it hard to figure out what is there and how to use it.

Overall, it seems than this packages is more research oriented rather than for practical use (e.g., thousands of models). 
So I am closing this issue again with a wont fix label.
